SPECS/bpftrace/bpftrace.spec

```diff
@@ -51,6 +48,7 @@ cd build
 
 %cmake \
     -DCMAKE_BUILD_TYPE=Release \
+    -DBUILD_SHARED_LIBS:BOOL=OFF \
```

Comment:
Chris, could you explain in a bit more detail why are these libraries special? In most cases we strongly prefer using shared libraries, because they make updates a lot easier, whereas every static lib usage is an additional complication, since now you have to rebuild all dependent packages. I know we have cases where there's no easy way around using static libs, so I just wanted to understand why we need to use them here.

Reply:
Wait, nevermind, I think I get it. We're not publishing any libraries from `bpftrace`, so there are no external specs depending on them. Is that it?

Reply:
Correct, we are not publishing any libraries from bpftrace and no external specs depend on them. And when we build with cmake -DBUILD_SHARED_LIBS=ON, rpm autodetects that libbpftrace.so shared libraries were produced during the build, so it dynamically autorequires them in the bpftrace package. We don't package them so we get package install errors like the following:

```
   # dnf install -y bpftrace
   Last metadata expiration check: 0:29:49 ago on Wed Feb  9 21:21:25 2022.
   Error:
   Problem: conflicting requests
     - nothing provides libast.so()(64bit) needed by bpftrace-0.13.0-1.cm2.x86_64
     - nothing provides libbpftrace.so()(64bit) needed by bpftrace-0.13.0-1.cm2.x86_64
   (try to add '--skip-broken' to skip uninstallable packages)
```

So either we build + package the shared libraries, knowing that no other package uses these shared libraries. Or we disable building of the additional shared libraries, which is what I elected to do here.

Note: I'm fine with either method, but figured since nothing else relied on those shared libraries, it didn't make sense to produce them.

Reply:
Makes sense and works for me. No need to overcomplicate it and we can always switch back to building shared libs, if needed.